source/server/active_tcp_listener.cc

```diff
@@ -284,6 +284,10 @@ void ActiveTcpListener::resumeListening() {
 
 void ActiveTcpListener::newConnection(Network::ConnectionSocketPtr&& socket,
                                       std::unique_ptr<StreamInfo::StreamInfo> stream_info) {
+  // populate the SNI into stream info, the network filters needn't do that again.
```

Comment:
Maybe something like `// Set the SNI as early as possible, this ensures that it gets set even if we end up not instantiating any network filters.`?

Reply:
got it